tiledb/sm/serialization/array_schema.cc

```diff
@@ -307,7 +307,15 @@ Status dimension_to_capnp(
 ByteVecValue tile_extent_from_capnp(
     const capnp::Dimension::TileExtent::Reader tile_extent_reader,
     Datatype dim_type) {
+  // When the datatype_size function returns 0, we know that the input datatype
+  // is invalid.
   auto coord_size = datatype_size(dim_type);
+  if (coord_size == 0) {
```

Comment:
Needs a comment that this return value signals an error, at the very least here, preferably also in the function declaration.